### PR TITLE
Added a small paragraph about environment modules

### PR DESCRIPTION
**Pull request by JensWehner**
These are still very popular on many HPC clusters.

- [x] I followed the [CONTRIBUTING guidelines](../blob/master/CONTRIBUTING.md).

Below, describe what this Pull Request adds:

Adds information about Linux environment modules

### CODE REVIEW

File: best_practices/language_guides/ccpp.md [hunk 1]
```diff
@@ -292,7 +294,8 @@ A third option that is sometimes used is to make separate "template implementati
 ### Testing
 Use [Google Test](https://github.com/google/googletest).
 It is light-weight, good and is used a lot.
-[Catch2](https://github.com/catchorg/Catch2) is also pretty good, well maintained and has native support in the CLion IDE.
+[Catch2](https://github.com/catchorg/Catch2) is also pretty good, well maintained and has native support in the 
+IDE.
```
Comment (egpbos) [inline best_practices/language_guides/ccpp.md:298]: I'm fine with generalizing to other IDEs, but did you check whether this is true for other IDEs than CLion? Main ones to check I guess would be Visual Studio, Eclipse, XCode.

File: best_practices/language_guides/ccpp.md [hunk 2]
```diff
@@ -104,6 +104,8 @@ If you want to keep things organized on systems where you use multiple versions
 You install each library in its own directory (`~/.local/pkg/<package>` for instance), then running `xstow` will create symlinks to the files in the `~/.local` directory (one above the XStow package directory).
 Using XStow in this way alows you to keep a single additional search path when compiling your next library.
 
+Another way to install multiple versions of the same software are [environment modules](https://modules.readthedocs.io/en/latest/) , which allow you to easily populate your `$PATH` and other environment variables to find the respective package. 
+
```
Comment (egpbos) [inline best_practices/language_guides/ccpp.md:108]: Modules is already mentioned in the first sentence of this section. Maybe you can merge these two paragraphs (yours and the first) into something like "Many pkgmans around. Standard on clusters: Modules. [your stuff]. Else [rest]"

Comment (egpbos) [reply best_practices/language_guides/ccpp.md:108]: Actually, also Easybuild (a Modules extension) is mentioned later on. Now that I look at it, I find the whole Package Management section to be rather messy and long :) Maybe we can make it less verbose at some point. In any case, for this PR I guess it makes sense to somehow connect these two paragraphs as well, up to you!

File: best_practices/language_guides/ccpp.md (hunk 1 above)
Comment (JensWehner) [reply best_practices/language_guides/ccpp.md:298]: actually I think that was an accidental edit. I revert that change.

**Pull request merged**